modules/phase_field/src/userobjects/EulerAngleUpdater.C

```diff
@@ -63,8 +66,10 @@ EulerAngleUpdater::initialize()
 
   for (unsigned int i = 0; i < grain_num; ++i)
   {
-    if (!_first_time && !_fe_problem.converged(_sys.number()))
+    // We dont want to use old angles on recovery
```

Comment:
why not if we are restarting the old angles?

Reply:
so old angles are for when a time step failed. To reset the angles to what they were before. 
So this logic is for failed time steps. 
But here we dont have a failed time step, we're loading a valid state.

besides angles_old is from time step n - 1, when angles should be from time step n, and are computed right above

Reply:
Can you add some additional doxygen to `_first_time_recovered` (and/or to other data members) that helpscommunicate what you just communicated here?

Reply:
should be good now